API/Controllers/SearchController.cs

```diff
@@ -1,119 +1,119 @@
-
-            SearchResultsResource searchResultsResource = new SearchResultsResource()
-            {
-                Results = searchResults.ToArray(),
-                Query = query,
-                Count = searchResults.Count(),
-                TotalCount = await searchService.SearchInternalProjectsCount(query, searchParams),
-                Page = searchParams.Page,
-                TotalPages =
-                await searchService.SearchInternalProjectsTotalPages(query, searchParams)
-            };
-
-            return Ok(searchResultsResource);
-        }
-
-    }
-
-}
+/*
+* Digital Excellence Copyright (C) 2020 Brend Smits
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published
+* by the Free Software Foundation version 3 of the License.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty
+* of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+* See the GNU Lesser General Public License for more details.
+*
+* You can find a copy of the GNU Lesser General Public License
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using API.Resources;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Models;
+using Services.Services;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace API.Controllers
+{
+
+    /// <summary>
+    ///     The controller that handles search requests
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class SearchController : ControllerBase
+    {
+
+        private readonly IMapper mapper;
+
+        private readonly ISearchService searchService;
+
+        /// <summary>
+        ///     Initialize a new instance of SearchController
+        /// </summary>
+        /// <param name="searchService"></param>
+        /// <param name="mapper"></param>
+        public SearchController(ISearchService searchService, IMapper mapper)
+        {
+            this.searchService = searchService;
+            this.mapper = mapper;
+        }
+
+        /// <summary>
+        ///     Search for projects
+        /// </summary>
+        /// <param name="query">The search query</param>
+        /// <param name="parameters"></param>
+        /// <returns>Search results</returns>
+        [HttpGet("internal/{query}")]
+        public async Task<IActionResult> SearchInternalProjects(string query,
+                                                                [FromQuery] RequestParamsResource parameters)
+        {
+            ProblemDetails problem = new ProblemDetails
+            {
+                Title = "Invalid search request."
+            };
+            if(query.Length < 0)

```

Comment:
A string can be null causing null pointer exceptions when calling the length

Reply:
```suggestion
            if(string.IsNullOrEmpty(query))

```